pkg/api/handlers/libpod/images.go

```diff
@@ -186,6 +193,118 @@ func ImagesImport(w http.ResponseWriter, r *http.Request) {
 }
 
 func ImagesPull(w http.ResponseWriter, r *http.Request) {
-	//TODO ...
-	utils.Error(w, "Something went wrong.", http.StatusInternalServerError, errors.New("/libpod/images/pull is not yet implemented"))
+	runtime := r.Context().Value("runtime").(*libpod.Runtime)
+	decoder := r.Context().Value("decoder").(*schema.Decoder)
+	query := struct {
+		Reference    string `schema:"reference"`
+		Credentials  string `schema:"credentials"`
+		OverrideOS   string `schema:"override-os"`
+		OverrideArch string `schema:"override-arch"`
+		TLSVerify    bool   `schema:"tls-verify"`
+		AllTags      bool   `schema:"all-tags"`
+	}{
+		TLSVerify: true,
+	}
+
+	if err := decoder.Decode(&query, r.URL.Query()); err != nil {
+		utils.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest,
+			errors.Wrapf(err, "failed to parse parameters for %s", r.URL.String()))
+		return
+	}
+
+	if len(query.Reference) == 0 {
```

Comment:
I find below more obvious:

if _, found := mux.Vars(r)["reference"]; !found {
...
}

Reply:
Sure, can do

Reply:
Just about to perform the change, I actually prefer the current way. It's faster as we don't need to look up and don't duplicate the schema string but can reference directly via the struct.